laterpay/application/Controller/Admin/TinyMCE.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/**
+ * Laterpay TinyMCE controller.
+ *
+ * Plugin Name: LaterPay
+ * Plugin URI: https://github.com/laterpay/laterpay-wordpress-plugin
+ * Author URI: https://laterpay.net/
+ */
+class LaterPay_Controller_Admin_TinyMCE extends LaterPay_Controller_Admin_Base {
+
+    /**
+     * @see LaterPay_Core_Event_SubscriberInterface::get_subscribed_events()
+     */
+    public static function get_subscribed_events() {
+
+        return array(
+            'laterpay_mce_buttons'                    => array(
+                array( 'laterpay_on_admin_view', 200 ),
+                array( 'laterpay_on_plugin_is_active', 200 ),
+                array( 'register_tinymce_button' ),
+            ),
+            'laterpay_mce_external_plugins'           => array(
+                array( 'laterpay_on_admin_view', 200 ),
+                array( 'laterpay_on_plugin_is_active', 200 ),
+                array( 'add_tinymce_button' ),
+            ),
+            'laterpay_admin_enqueue_styles_post_edit' => array(
+                array( 'laterpay_on_admin_view', 200 ),
+                array( 'localize_script', 50 ),
+            ),
+            'laterpay_admin_enqueue_styles_post_new'  => array(
+                array( 'laterpay_on_admin_view', 200 ),
+                array( 'localize_script', 50 ),
+            ),
+        );
+
+    }
+
+    /**
+     * To add laterpay short code generator dropdown in TinyMCE.
+     *
+     * @param LaterPay_Core_Event $event
+     *
+     * @return void
+     */
+    public function register_tinymce_button( LaterPay_Core_Event $event ) {
+
+        $result = $event->get_result();
+
+        $result = array_merge(
+            $result,
+            [
+                'laterpay_shortcode_generator',
+            ]
+        );
+
+        $event->set_result( $result );
+
+    }
+
+    /**
+     * To register script for custom button for shortcode generator dropdown.
+     *
+     * @param LaterPay_Core_Event $event
+     *
+     * @return void
+     */
+    public function add_tinymce_button( LaterPay_Core_Event $event ) {
+
+        $result = $event->get_result();
+
+        $result['laterpay_shortcode_generator'] = sprintf( '%slaterpay-backend-shortcode-generator.js', $this->config->js_url );
```

Comment:
The script should only load on Post Types allowed for sale, else it would throw an error on load.

`in_array( $post_type, $this->config->get( 'content.enabled_post_types' )`